cql3/statements/modification_statement.cc

```diff
@@ -349,6 +387,13 @@ modification_statement::execute_with_condition(query_processor& qp, service::que
     auto token = request->key()[0].start()->value().as_decorated_key().token();
 
     auto shard = service::storage_proxy::cas_shard(*s, token);
+
+    if (utils::get_local_injector().is_enabled("forced_bounce_to_shard_counter")) {
+        if constexpr (std::same_as<utils::error_injection_type, utils::error_injection<true>>) {
```

Comment:
Why do we need this 'if'? If error_injection is disabled during compilation `is_enable` always returns false, so everything here should be compiled out anyway.